unified-runtime/source/adapters/level_zero/v2/queue_immediate_out_of_order.cpp

```diff
@@ -0,0 +1,189 @@
+//===--------- queue_immediate_in_order.cpp - Level Zero Adapter ---------===//
+//
+// Copyright (C) 2024 Intel Corporation
+//
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "queue_immediate_out_of_order.hpp"
+#include "../common/latency_tracker.hpp"
+#include "ur.hpp"
+
+namespace v2 {
+
+// Helper function to intialize std::array of command list manager.
+// This is needed because command list manager does not have a default
```

Comment:
tbh, I'm tempted to ask you to use a vector to avoid these templated functions :)